packages/core/src/new-api/deploy.ts

```diff
@@ -0,0 +1,57 @@
+import { Deployer } from "./internal/deployer";
+import { FileDeploymentLoader } from "./internal/deployment-loader/file-deployment-loader";
+import { MemoryDeploymentLoader } from "./internal/deployment-loader/memory-deployment-loader";
+import { EthersChainDispatcher } from "./internal/execution/transactions/chain-dispatcher";
+import { TransactionServiceImplementation } from "./internal/execution/transactions/transaction-service";
+import { Adapters } from "./types/adapters";
+import { ArtifactResolver } from "./types/artifact";
+import { DeploymentResult } from "./types/deployer";
+import { IgnitionModuleResult, ModuleParameters } from "./types/module";
+import { IgnitionModuleDefinition } from "./types/module-builder";
+
+/**
+ * Deploy an IgnitionModule to the chain
+ *
+ * @beta
+ */
+export function deploy({
```

Comment:
This is the new entry point from plugin. We can use deployer from here and core tests, while not exposing more than the signature of this method.

Happy to split/refactor/classify this function, but I wanted remove journal/deployer/transactionservice from being exported.